airflow/providers/google/cloud/operators/dataproc.py

```diff
@@ -3055,6 +3055,13 @@ def execute(self, context: Context):
                     self.log.info("Batch %s created", self.batch_id)
 
                 else:
+                    DataprocBatchLink.persist(
```

Comment:
It looks like this block is not called in the deferrable mode (which should be triggered by `self.defer()`), but in the sync mode if `self.asynchronous` is not `True`. 

Could you add some unit tests showing that the link is persisted in the deferrable mode?

Reply:
I see that the same question has already been asked. Closing my comment.